### PR TITLE
Dependency upgrades

### PR DESCRIPTION
**Pull request by kinetifex**
Contribution package upgrades, related to this discussion: https://github.com/gulpjs/gulp/discussions/2590

Looks like the work to update `devDependencies` was being tackled in https://github.com/gulpjs/vinyl-fs/pull/322.

### CODE REVIEW

**Comment by phated** [comment]
Thanks, but dependency upgrades get handled by our updater tool. They are strictly semver managed.

File: package.json [hunk 1]
```diff
@@ -27,21 +27,21 @@
...
     "lazystream": "^1.0.0",
-    "lead": "^1.0.0",
+    "lead": "^3.0.0",
     "object.assign": "^4.0.4",
-    "pumpify": "^1.3.5",
-    "readable-stream": "^2.3.3",
+    "pumpify": "^2.0.1",
+    "readable-stream": "^3.6.0",
     "remove-bom-buffer": "^3.0.0",
     "remove-bom-stream": "^1.2.0",
```
Comment (kinetifex) [inline package.json:40]: So, everything upgraded smoothing expect I have run into some issues with [remove-bom-stream v2](https://github.com/gulpjs/remove-bom-stream) which was published not too long ago.

I haven't delved too deep into it yet, but wanted to reach out if there's some early guidance for tackling this @phated? Is kinda the goal to move from `through2` to `streamx` across-the-board (per https://github.com/gulpjs/remove-bom-stream/issues/4)?

Comment (phated) [reply package.json:40]: Yes, almost the entirety of the gulp 5 holdup is finding someone to do the streamx updates across all projects.

File: test/dest.js [hunk 2]
```diff
@@ -945,6 +945,7 @@ describe('.dest()', function() {
     pipe([
       from.obj([file]),
       destStream,
+      slowCount(1), // Verify and avoid race condition w/ assert check
```
Comment (kinetifex) [inline test/dest.js:948]: I seemed to start having failing assets after upgrading `readable-stream` to v3, including the packages that had v2 as a sub dependency. It feels like a race condition with these tests that I left code comments on. Does this test fix make sense?

File: test/integration.js [hunk 3]
```diff
@@ -55,10 +52,15 @@ describe('integrations', function() {
     }
 
     pipe([
-      vfs.src(inputGlob, { buffer: false }),
+      vfs.src(inputGlob, { highWaterMark: expectedCount }),
```
Comment (kinetifex) [inline test/integration.js:55]: I could not find `buffer` in use. `highWaterMark` is not a documented option, but is used in code and other tests, so I went with it.

File: package.json (hunk 1 above)
Comment (kinetifex) [reply package.json:40]: I see - would you be open to an intermediate release for some of these upgrades to `vinyl-fs` for now? Maybe leave the streamx work out for now but catch up on some of these other lagging dependencies. I left some other comment questions, but so far it doesn't seem like anything would constitute a major breaking change.

Comment (phated) [reply package.json:40]: Every major causes a major. All of the ones in the gulpjs org dropped node support. Everything leads to gulp 5 with no options for "intermediate"

**Pull request closed without merging**